validator/server/src/main/java/org/ethereum/beacon/validator/api/ValidatorRest.java

```diff
@@ -0,0 +1,389 @@
+                time = observableBeaconState.getLatestSlotState().getGenesisTime();
+              }
+            });
+
+    Flux.from(syncManager.getSyncStatusStream())
+        .subscribe(
+            syncStatus -> {
+              shortSync = SyncManager.SyncMode.Short.equals(syncStatus.getSyncMode());
+              syncingResponse =
+                  new SyncingResponse(
+                      syncStatus.isSyncing(),
+                      syncStatus.getStart() == null
+                          ? BigInteger.ZERO
+                          : syncStatus.getStart().toBI(),
+                      syncStatus.getCurrent() == null
+                          ? BigInteger.ZERO
+                          : syncStatus.getCurrent().toBI(),
+                      syncStatus.getBestKnown() == null
+                          ? BigInteger.ZERO
+                          : syncStatus.getBestKnown().toBI());
+            });
+  }
+
+  @Override
+  HttpServer addRoutes(HttpServer httpServer) {
+    return httpServer.route(
+        routes ->
+            routes
+                .get("/node/version", doJsonResponse(this::produceVersionResponse))
+                .get("/node/genesis_time", doJsonResponse(this::produceGenesisTimeResponse))
+                .get("/node/syncing", doJsonResponse(this::produceSyncingResponse))
+                .get(
+                    "/validator/duties",
+                    processGetRequestImpl(this::produceValidatorDutiesResponse))
+                .get("/validator/block", processGetRequestImpl(this::produceValidatorBlockResponse))
+                .post("/validator/block", processPostRequestImpl(this::acceptBlockSubmit))
+                .get(
+                    "/validator/attestation",
+                    processGetRequestImpl(this::produceAttestationResponse))
+                .post(
+                    "/validator/attestation", processPostRequestImpl(this::acceptAttestationSubmit))
+                .get("/node/fork", doJsonResponse(this::produceForkResponse)));
+  }
+
+  private String produceVersionResponse() {
+    if (version == null) {
+      Properties props = loadResources("rest-server.properties", this.getClass().getClassLoader());
+      final String version = props.getProperty("versionNumber");
+      this.version = String.format("Beacon Chain Java %s", version);
+    }
+    return version;
+  }
+
+  private Properties loadResources(final String name, final ClassLoader classLoader) {
+    try {
+      final Enumeration<URL> systemResources =
+          (classLoader == null ? ClassLoader.getSystemClassLoader() : classLoader)
+              .getResources(name);
+      Properties props = new Properties();
+      props.load(systemResources.nextElement().openStream());
+
+      return props;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private String produceSyncingResponse() {
+    try {
+      return mapper.writeValueAsString(syncingResponse);
```

Comment:
It would be handy if this mapping was segregated and put into a middleware pipeline.